src/conf.cpp

```diff
@@ -39,244 +39,138 @@ struct section {
     struct section *next;
 };
 
-ConfFile::~ConfFile()
+ConfFile::ConfFile(const char *filename)
+    : _filename(filename)
+    , _entry(nullptr)
+    , _last_section(nullptr)
+    , _n(0)
+    , _line(0)
```

Comment:
Isn't this just changing the place in which information is kept in memory? On mavlink-router this is ConfFile. Here you are using ConfFile just to parse the file and storing information in your new Settings class... and you are storing it as a map of strings.

I think it would make much more sense to have it restructured, both here and on mavlink-router to do like this:

1. Open file in memory with mmap
2. Run through it once and keep track of the offset of every section. Don't parse anything else;
3. Users then request the settings of a particular section passing in a table with what it's expecting and how to convert it to the proper format and where to store the result
a) ConfParser::parse_section("xyz", tbl), in which tble is a table with { key, val_parse_func, ptr_storage }
b) ConfParser will quickly locate the section (since it had already stored the offset for it) and for each key call the correspondent val_parse_func passing in the ptr_storage and key

This way we don't keep duplicating the strings and rather store the info in the format they are going to be used. I.e. if it's a bool, we store it as a bool. And we can store it directly in the place that will make use of it.

Of course we can provide helper functions like "parse_bool_value()" that translates "yes|no|true|false" to true|false.
Thoughts?